packages/flutter/lib/src/material/app.dart

```diff
@@ -628,15 +659,20 @@ class _MaterialAppState extends State<MaterialApp> {
         onGenerateInitialRoutes: widget.onGenerateInitialRoutes,
         onUnknownRoute: widget.onUnknownRoute,
         builder: (BuildContext context, Widget child) {
-          // Use a light theme, dark theme, or fallback theme.
+          // Resolve which theme to use based on brightness and high contrast.
           final ThemeMode mode = widget.themeMode ?? ThemeMode.system;
+          final ui.Brightness platformBrightness = MediaQuery.platformBrightnessOf(context);
+          final bool useDarkTheme = mode == ThemeMode.dark
+              || (mode == ThemeMode.system && platformBrightness == ui.Brightness.dark);
+          final bool highContrast = MediaQuery.highContrastOf(context);
           ThemeData theme;
-          if (widget.darkTheme != null) {
-            final ui.Brightness platformBrightness = MediaQuery.platformBrightnessOf(context);
-            if (mode == ThemeMode.dark ||
-              (mode == ThemeMode.system && platformBrightness == ui.Brightness.dark)) {
-              theme = widget.darkTheme;
-            }
+
+          if (useDarkTheme && highContrast) {
+            theme = widget.highContrastDarkTheme;
+          } else if (useDarkTheme) {
+            theme = widget.darkTheme;
+          } else if (highContrast) {
+            theme = widget.highContrastTheme;
           }
           theme ??= widget.theme ?? ThemeData.fallback();
```

Comment:
The extra level of indirection provided by `ThemeData.fallback()` doesn't provide much value. Maybe just use ThemeData.light() here. It's a little clearer.

Reply:
Done